packages/caip/src/adapters/coingecko/utils.ts

```diff
@@ -147,8 +163,11 @@ export const parseData = (coins: CoingeckoCoin[]): AssetMap => {
 export const writeFiles = async (data: AssetMap) => {
   await Promise.all(
     Object.entries(data).map(async ([chainId, assets]) => {
-      const path = `./src/adapters/coingecko/generated/${chainId}/adapter.json`.replace(':', '_')
-      await fs.promises.writeFile(path, JSON.stringify(assets))
+      const dirPath = `./src/adapters/coingecko/generated/${chainId}`.replace(':', '_')
+      if (!fs.existsSync(dirPath)) {
+        fs.mkdirSync(dirPath);
+      }
+      await fs.promises.writeFile(`${dirPath}/adapter.json`, JSON.stringify(assets))
```

Comment:
If the caip dir/file doesn't exist yet the `generate` script fails. This addresses that problem